src/main/java/edu/wpi/grip/core/SocketHints.java

```diff
@@ -0,0 +1,185 @@
+                                                                       final Number low, final Number high) {
+            return createNumberSocketHintBuilder(identifier, number, new Number[]{low, high}).view(SocketHint.View.SPINNER).build();
+        }
+
+        public static SocketHint<Number> createNumberSpinnerSocketHint(final String identifier, final Number number,
+                                                                       final Domain domain) {
+            return createNumberSocketHintBuilder(identifier, number, domain.getDomain()).view(SocketHint.View.SPINNER).build();
+        }
+
+        public static SocketHint<Number> createNumberSpinnerSocketHint(final String identifier, final Number number) {
+            return createNumberSocketHintBuilder(identifier, number).view(SocketHint.View.SPINNER).build();
+        }
+
+        public static SocketHint<List> createNumberListRangeSocketHint(final String identifier, final Number low, final Number high) {
+            return createNumberListSocketHintBuilder(identifier, new Number[]{low, high})
+                    .view(SocketHint.View.RANGE)
+                    .build();
+        }
+
+        public static SocketHint<List> createNumberListRangeSocketHint(final String identifier, final Domain domain) {
+            return createNumberListSocketHintBuilder(identifier, domain.getDomain())
+                    .view(SocketHint.View.RANGE)
+                    .build();
+        }
+    }
+
+    /**
+     * These {@link SocketHint SocketHints} should only be used in {@link InputSocket InputSockets}
+     */
+    public static final class Outputs {
+        private Outputs() { /* no op */ }
+
+        public static SocketHint<Mat> createMatSocketHint(final String identifier) {
+            return Inputs.createMatSocketHint(identifier, true);
+        }
+
+        public static SocketHint<Point> createPointSocketHint(final String identifier) {
+            return Inputs.createPointSocketHint(identifier, true);
+        }
+
+        public static SocketHint<Size> createSizeSocketHint(final String identifier) {
+            return Inputs.createSizeSocketHint(identifier, true);
+        }
+
+        public static SocketHint<Boolean> createBooleanSocketHint(final String identifier, boolean defaultValue) {
+            return new SocketHint.Builder(Boolean.class)
+                    .identifier(identifier)
+                    .initialValue(defaultValue)
+                    .publishable(true).build();
+        }
+
+        public static SocketHint<Number> createNumberSocketHint(final String identifier, Number defaultValue) {
+            return new SocketHint.Builder(Number.class)
+                    .identifier(identifier)
+                    .initialValue(defaultValue)
+                    .publishable(true).build();
+        }
+    }
+
+    public static <T extends Enum<T>> SocketHint<T> createEnumSocketHint(final String identifier,
```

Comment:
I don't see why you need to

    ... createEnumSocketHint(String id, Class<T> type) {
            return new SocketHint.Builder()
                    .identifier(id)
                    .domain(type.getEnumConstants())
                    .initialValue(type.getEnumConstants()[0])
                    .view(SocketHint.View.SELECT)
                    .build();
    }

(And probably a version with an explicit default value too, but you get the point)

That way, we can really easily make a socket hint automatically for any enum:

    private final SocketHint<Whatever> whateverHint = SocketHints.createEnumSocketHint("something", Whatever.class);